### PR TITLE
add create command to zimlet cli to quickly generate new zimlets from templates stored in github

### PR DESCRIPTION
**Pull request by billneff79**
Borrowed heavily from [preact-cli](https://github.com/developit/preact-cli) for this one.

I created a simply template at [billneff79/zimlet-default-template](https://github.com/billneff79/zimlet-default-template).  You can test this by doing

```sh
zimlet-cli $ git checkout zimlet-create
zimlet-cli $ npm install && npm run build && npm link
zimlet-cli $ cd ~/repos # or whatever your repos directory is
repos $ zimlet create billneff79/zimlet-default-template my-zimlet

# or
repos $ zimlet create   # follow prompts 
```

### CODE REVIEW

File: src/commands/package.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import fs from 'fs';
```
Comment (billneff79) [inline src/commands/package.js]: no real changes here - this was just cut/pasted out of `index.js`

File: src/commands/create.js [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+			// Write default "scripts" if none found
+			pkgData.scripts = pkgData.scripts || (await addScripts(pkgData, target, isYarn));
+		}
+		else {
+			warn('Could not locate `package.json` file!');
+		}
+
+		// Update `package.json` key
+		if (pkgData) {
+			spinner.succeed().start('Updating `name` within `package.json` file');
```
Comment (pl12133) [inline src/commands/create.js]: I would like to see this include the generated name so the user is sure they generated it correctly, e.g.

```
'Updating `name` to `my-package` within `package.json` file'
```

File: src/commands/create.js [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+				}
+			}
+		});
+
+		if (keeps.length) {
+			let dict = new Map();
+
+			// TODO: concat author-driven patterns
+			//globally replace instances of {{name}} with value of name arg
+			['name'].forEach(str => {
```
Comment (pl12133) [inline src/commands/create.js]: The template replacement seems to be failing, I'm seeing a couple untransformed `{{name}}` strings in a newly generated project:

```sh
$: grep -R -e '{{name}}' src
src/constants/index.js:export const SLUG = '{{name}}';
src/index.js:                   <Text id="{{name}}.menuItem" />
src/components/app/style.less:.{{name}} {
src/intl.json:  "{{name}}": {
```

File: src/lib/setup.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+		}
+		catch (e) {
+			gitEmail = defaultGitEmail;
+		}
+
+		try {
+			gitUser = (await spawn('git', ['config', 'user.name'])).toString();
+		}
+		catch (e) {
+			gitUser = defaultGitUser;
```
Comment (pl12133) [inline src/lib/setup.js]: Is it too verbose to warn the user when we fallback to default git email/username values?

File: src/commands/create.js (hunk 3 above)
Comment (billneff79) [reply src/commands/create.js]: - [ ] fix not all `{{name}}` references being replace

File: src/commands/create.js (hunk 2 above)
Comment (billneff79) [reply src/commands/create.js]: - [ ] add actual name to replacement status message

File: src/lib/setup.js (hunk 4 above)
Comment (billneff79) [reply src/lib/setup.js]: no, I think that is a good warning to have.  I'll put it in

**Comment by billneff79** [comment]
I think I've addressed all of your comments @pl12133

**Pull request merged**